MMX on RDKB

MuteSpirit opened the pull request:
Yocto recipes for MMX core packages
prplMesh customer and backend recipes will be in [meta-prplmesh](https://gitlab.com/prpl-foundation/prplmesh/meta-prplmesh) layer

Commit:
Add conf/layer.conf

Set only Dunfell support
   because this layer will be tested with that Yocto version

Set layer priority is 9

Enable
    BB_DANGLINGAPPENDS_WARNONLY = "1"
to be able build with Yocto and RDKB

Signed-off-by: Ivan Efimov <i.efimov@inango-systems.com>

Commit:
Recipe for microxml library

Used strict version "2015-03-18" of microxml library

Signed-off-by: Ivan Efimov <i.efimov@inango-systems.com>

Commit:
Recipe for LuaXML (Lua bindings for LibXML2 library)

Used strict "101012" version of LuaXML

TODO: fix license (CLOSED is used now)

Signed-off-by: Ivan Efimov <i.efimov@inango-systems.com>

Commit:
Recipe for klish

The klish is a framework for implementing a CISCO-like CLI on a
UNIX systems

Use strict version 2.1.4

Signed-off-by: Ivan Efimov <i.efimov@inango-systems.com>

Commit:
Recipe for libing-gen-utils

libing-gen-utils is a number of C and Lua useful general utilities
and modules which may be used in MMX backends

Use version 2.0.1.1

Signed-off-by: Ivan Efimov <i.efimov@inango-systems.com>

Commit:
Recipe for libmmx-backapi

libmmx-backapi is library for communication between MMX EP and
MMX backend made as a deamon

Use version 2.0.1.1

Signed-off-by: Ivan Efimov <i.efimov@inango-systems.com>

Commit:
Recipe for libmmx-frontapi

libmmx-frontapi is a library to communicate between MMX frontends
and MMX EntryPoint

Use version 2.0.1.1

Signed-off-by: Ivan Efimov <i.efimov@inango-systems.com>

Commit:
Recipe for MMX EntryPoint (mmx-ep)

MMX EntryPoint is a core of MMX system

It store data model, harvest data from backends and return data
to MMX frontends (CLI, WEB, ...)

Use version 2.0.1.1

Use files from two repo: mmx-ep (sources) and feed-mmx (init scripts)

Signed-off-by: Ivan Efimov <i.efimov@inango-systems.com>

Commit:
Recipe for MMX CLI (mmx-cli)

CLI frontend for MMX

Based on klish

Use version 2.0.1.1

Signed-off-by: Ivan Efimov <i.efimov@inango-systems.com>

Commit:
Set lububox and ubus versions from meta-openwrt as preferred

For example Soft.At.Home meta layer for Ambiorix is oriented on
the latest version of libubox, ubus, Lua and so on

Force using libubox and ubus from meta-openwrt to avoid issues
with compatibility

Signed-off-by: Ivan Efimov <i.efimov@inango-systems.com>

Commit:
Package group for MMX

Add packagegroup for MMX with packages required for MMX CLI:
* ubus
* mmx-ep
* mmx-cli
* libubox

On add MMX Web support this list must be extended

Signed-off-by: Ivan Efimov <i.efimov@inango-systems.com>

Commit:
Disable "meta-openwrt" busybox patches and base-files "includeopenwrt" feature

Busybox patches are conflict with busybox in RDKB OE 3.1
Disable them.

base-files "includeopenwrt" feature cause conflict of installation
the same files by "base-files" and several other packages,
for example, iptables2/procs

Signed-off-by: Ivan Efimov <i.efimov@inango-systems.com>

Commit:
iwinfo bbappend: adopt to use Lua 5.1

Add dependency on "lua5.1" and include dir to .../include/lua5.1/

Signed-off-by: Ivan Efimov <i.efimov@inango-systems.com>

Commit:
uci bbappend: add libuci.so into FILES

Add bbappend to "uci" recipe from "meta-openwrt" to fix Yocto
warning that "/usr/lib/libuci.so" is not in any package

Signed-off-by: Ivan Efimov <i.efimov@inango-systems.com>

Commit:
ubus bbappend: systemd unit for ubusd

Systemd service should just run ubusd daemon in background mode

Signed-off-by: Ivan Efimov <i.efimov@inango-systems.com>

Commit:
mmx-web: recipe for MMX WEB

MMX WEB contains:
* Lua modules for LuCI
* htm files for views (vector, matrix, tableview, field)
* MMX WEB JavaScript

Signed-off-by: Ivan Efimov <i.efimov@inango-systems.com>

Commit:
packagegroup-mmx: add MMX WEB related packages

Add into MMX package group:
* mmx-web
* luci
* uhttpd
* ubus
* rpcd
* procd-base-files

Signed-off-by: Ivan Efimov <i.efimov@inango-systems.com>

Commit:
procd-base-files recipe

Add a set of scripts (rc.common, ...) to start OpenWrt services
on RDKB, for example, uhttpd or rpcd

Signed-off-by: Ivan Efimov <i.efimov@inango-systems.com>

Commit:
lucihttp recipe

HTTP utility library with Lua binding

LuCI need this this library to work

Signed-off-by: Ivan Efimov <i.efimov@inango-systems.com>

Commit:
LuCi recipe bbappend

Additionally to luci recipe:
* install libs/luci-lib-base
* install luci_mod_network
* install luci_lib_base

Add patch to fix compilation "pluralParseFree" function into
parser.so library of luci-base module

Add explicit runtime dependency on "lucihttp" library

Signed-off-by: Ivan Efimov <i.efimov@inango-systems.com>

Commit:
luaxml: fix license from CLOSED to MIT

License "CLOSED" has been used temporary to forward with recipe
Real license is MIT

Set MIT as luaxml license.
Use tail of "src/LuaXML_${PV}/readme.txt" as license location

Signed-off-by: Ivan Efimov <i.efimov@inango-systems.com>

Commit:
luaxml: fix host paths usage during cross-compilation

There is a compilation warning that "/usr/include" folder is used
for compilation.

LuaXML makefile defines include directory using STAGING_DIR variable
but there is no default value for that, neigther in Makefile nor in
recipe.

Define STAGING_DIR to avoid using host paths during compilation

Signed-off-by: Ivan Efimov <i.efimov@inango-systems.com>

Commit:
luaxml: add LUAPATH usage

RDKB install Lua modules in path containing Lua version,
   i.e. /usr/lib/lua/5.1

Defind LUAPATH in recipe to follow this policy

Signed-off-by: Ivan Efimov <i.efimov@inango-systems.com>

Commit:
libing-gen-utils: use LUAPATH during installation

Redefine LUAPATH variable and use in do_install to put Lua
modules
    into        /usr/lib/lua/5.1/mmx
        instead of  /usr/lib/lua/mmx

Signed-off-by: Ivan Efimov <i.efimov@inango-systems.com>

Commit:
libmmx-frontapi: use LUAPATH during installation

Redefine LUAPATH variable and use in do_install to put Lua
modules
    into        /usr/lib/lua/5.1/mmx
    instead of  /usr/lib/lua/mmx

Signed-off-by: Ivan Efimov <i.efimov@inango-systems.com>

Commit:
mmx-web: reconfigure LuCI and uhttpd for MMX purposes

Switch "mmx-web" to commit
    commit d85c535323cd06a2f6b20acac9128509fd3eede3
    Merge: ac41d21 a97286a
    Author: VladyslavTupikin
    <61505283+VladyslavTupikin@users.noreply.github.com>
    Date:   Tue Mar 2 10:29:36 2021 +0200

        Merge pull request #36 from
        InangoSystems/hotfix/dispatcher_lua_patch

        mmx-web: Fix patching dispatcher.lua

Modify do_install to
* copy Lua modules/files into /usr/lib/lua/5.1/...
* install patched dispatcher.lua
    (with support pages with action "call")

Add next patches for MMX Web uci-defaults scripts:
1) add LUAPATH support to mmx web uci defaults script
2) add luci.template.viewdir="/usr/lib/lua/5.1/luci/view/"
3) fix luci location in uhttpd settings:
   from: /cgi-bin/luci=/usr/lib/lua/luci/sgi/uhttpd.lua
   to:   /cgi-bin/luci=/usr/lib/lua/5.1/luci/sgi/uhttpd.lua
4) set uhttpd listen ports:
   http:  2080
   https: 2443

Signed-off-by: Ivan Efimov <i.efimov@inango-systems.com>

Commit:
README.md: add manuals how to build and run MMX

Attention:
* meta-openwrt uses "master" branch and need patch to "conf/layer.conf"
  to use in Dunfell
  Migration to "dunfell" branch - in separate task
* Besides "meta-mmx" you need also layer with customer package
  i.e. "meta-prplmesh" with "inango-prplmesh-en" package
  Otherwise MMX EP will not start and CLI/WEB won't work

Signed-off-by: Ivan Efimov <i.efimov@inango-systems.com>

Jonrion commented on recipes-extended/lucihttp/lucihttp_0.1.bb:
```diff
@@ -0,0 +1,105 @@
+
+PACKAGECONFIG[lua] = "-DBUILD_LUA=ON -DLUAPATH=${LUAPATH},-DBUILD_LUA=OFF,lua5.1,lua5.1"
+PACKAGECONFIG[tests] = "-DBUILD_TESTS=ON,-DBUILD_TESTS=OFF,,"
+#OECMAKE_EXTRA = " \
```
@MuteSpirit any reason to keep these 3 lines?
I expect tests being build after adding ```PACKAGECONFIG_append = " tests"```

MuteSpirit replied:
I've removed comment on line 99.

Line 97 is needed because it's used here
```
PACKAGECONFIG ??= "lua"
```

Line 98 is not needed right now but it's common practice to put possible options for PACKAGECONFIG but doesn't use in main recipe and allow do that in bbappends at external meta layers

MuteSpirit replied:
This recipe has been removed

ipodovalov commented on README.md:
```diff
@@ -89,10 +89,30 @@ repo sync -j$(nproc) --no-clone-bundle
+Fetch meta layer with customer package, i.e. [meta-prplmesh](https://gitlab.com/prpl-foundation/prplmesh/meta-prplmesh)
+See details inside README.md of that repo
+
+Apply next patch to allow use it in Dunfell
```
Could you, please, specify the repo where we must apply this patch (meta-openwrt).
For me, it wasn't obvious at first sight and required time to understand. Something like that:
`Apply the next patch in meta-openwrt layer to allow use it in Dunfell`
It is just a suggestion...

MuteSpirit replied:
Agree. Fixed

ipodovalov commented on recipes-core/iwinfo/iwinfo_git.bbappend:
```diff
@@ -0,0 +1,71 @@
+#
+################################################################################
+DEPENDS += "lua5.1"
+LDFLAGS_append = "-L${STAGING_LIBDIR}"
```
You must control all spacing when you use the override syntax. As for me
`LDFLAGS_append = "-L${STAGING_LIBDIR}"`
should be
`LDFLAGS_append = " -L${STAGING_LIBDIR}"`

MuteSpirit replied:
Agree. Fixed

ipodovalov commented on recipes-core/packagegroups/packagegroup-mmx.bb:
```diff
@@ -0,0 +1,93 @@
+
+inherit packagegroup
+
+RDEPENDS_${PN} = "\
```
I think it is desirable to use one of these two styles:
```
FOO = "\
       bar \
       done \
       "
```
or 
```
FOO = "\
    bar \
    done \
"
```
and not combine them as you do.

MuteSpirit replied:
I've chosen 2nd style and fixed exist recipes in this meta layer

ipodovalov commented on recipes-core/iwinfo/iwinfo_git.bbappend:
```diff
@@ -0,0 +1,71 @@
+# - professional sub-contract and customization services
+#
+################################################################################
+DEPENDS += "lua5.1"
```
To build iwinfo I had to remove 0001-Makefile-LDFLAGS-set-liblua5.1-for-lua-lib.patch
from SRC_URI:
`SRC_URI_remove += "file://0001-Makefile-LDFLAGS-set-liblua5.1-for-lua-lib.patch"`
Am I doing something wrong during my build?

MuteSpirit replied:
I had to do the same
I see such changes in branch `feature/PPM-1154-mmx-on-rdkb` of `meta-prplmesh` layer
https://gitlab.com/prpl-foundation/prplmesh/meta-prplmesh/-/blob/feature/PPM-1154-mmx-on-rdkb/recipes-core/iwinfo/iwinfo_git.bbappend

And i agree that removing patch must be done in `meta-mmx` instead

arnout commented on recipes-core/procd-base-files/procd-base-files_git.bb:
```diff
@@ -0,0 +1,107 @@
+
+do_install() {
+    install -d ${D}${sysconfdir}
+    install -Dm 0755 ${BF}/etc/rc.common ${D}${sysconfdir}/rc.common
```
(nitpick) The -D option creates the directory, so it's redundant with the preceding `install -d`.

MuteSpirit replied:
Fixed

arnout commented on recipes-core/procd-base-files/procd-base-files_git.bb:
```diff
@@ -0,0 +1,107 @@
+    install -Dm 0755 ${BF}/etc/rc.common ${D}${sysconfdir}/rc.common
+
+    install -d ${D}${base_libdir}
+    install -Dm 0644 ${BF}/lib/functions/service.sh ${D}${base_libdir}/functions/service.sh
```
Same here.

MuteSpirit replied:
Fixed

arnout commented on README.md:
````diff
@@ -104,18 +124,49 @@ BBLAYERS =+ "${RDKROOT}/meta-openwrt"
-Build image
 ```bash
-bitbake rdk-generic-broadband-image
+rpcd -s /var/run/ubus/ubus.sock -t 30 &
````
Why isn't there a service file to start this?

MuteSpirit replied:
I've added systemd unit for rpcd

Commit:
rpcd append: add systemd unit for rpcd daemon

Add systemd unit for rpcd (OpenWrt UBUS RPC service)
rpcd.service depends on ubusd systemd unit

Replace in README.md the step of run rpcd to step of check that
it's running

Signed-off-by: Ivan Efimov <i.efimov@inango-systems.com>

Commit:
Fix code style for multiline strings

Chosend style of multiline string definition is

FOO = "\
    bar \
    done \
"

Signed-off-by: Ivan Efimov <i.efimov@inango-systems.com>

ipodovalov commented on conf/layer.conf:
```diff
@@ -0,0 +1,84 @@
+
+BBPATH .= ":${LAYERDIR}"
+
+BBFILES += "${LAYERDIR}/recipes-*/*/*.bb \
```
```
BBFILES += "\
    ${LAYERDIR}/recipes-*/*/*.bb \
    ${LAYERDIR}/recipes-*/*/*.bbappend \
"
```
Please, amend your recent commit (Fix code style for multiline strings).

MuteSpirit replied:
fixed

kisyan89 commented on README.md:
````diff
@@ -104,18 +124,49 @@ BBLAYERS =+ "${RDKROOT}/meta-openwrt"
+/usr/sbin/uhttpd -f -h /www -r OpenWrt -x /cgi-bin -l /cgi-bin/luci -L /usr/lib/lua/5.1/luci/sgi/uhttpd.lua -t 60 -T 30 -k 20 -A 1 -n 3 -N 100 -R -p 0.0.0.0:2080 -p [::]:2080 &
+```
+
+Grand access to MMX WebUI from outside
````
Grand -> Grant

MuteSpirit replied:
fixed

kisyan89 commented on recipes-core/iwinfo/iwinfo_git.bbappend:
```diff
@@ -0,0 +1,72 @@
+################################################################################
+#
+# iwinfo_git.bb
```
Should be iwinfo_git.bbappend ?

MuteSpirit replied:
fixed

kisyan89 commented on recipes-core/procd-base-files/procd-base-files_git.bb:
```diff
@@ -0,0 +1,108 @@
+SRCREV = "6acc48c7a2faac48c534b8a5516500c270550a9e"
+
+S = "${WORKDIR}/git"
+PD = "${S}/openwrt/package/system/procd/files"
```
what is var PD and var BF ?

MuteSpirit replied:
That is subpathes to ProcD files (PD) and Base Files (BF) folders within `${S}`
Their goal is using short paths in `do_install` task

MuteSpirit replied:
Please resolve conversation if my explanation is enough

kisyan89 replied:
Looks like I can't resolve conversations, please resolve by yourself

kisyan89 commented on recipes-core/uci/uci_git.bbappend:
```diff
@@ -0,0 +1,72 @@
+################################################################################
+#
+# uci_git.bb
```
Should be uci_git.bbappend ?
Please check other bbappends as well

MuteSpirit replied:
fixed

kisyan89 commented on recipes-mmx/klish/klish_2.1.4.bb:
```diff
@@ -0,0 +1,86 @@
+
+S = "${WORKDIR}/klish-${PV}"
+
+FILES_${PN} += "/usr/lib"
```
Does not it get to FILES_${PN} by default as declared in `meta/conf/bitbake.conf` ?

Ranihilator replied:
export exec_prefix = "/usr"
export libdir = "${exec_prefix}/lib"

MuteSpirit replied:
This is case when Yocto doesn't add so file from that folder because it's a plugin so library
If remove this line then bitbake will print next error message:
```
WARNING: klish-2.1.4-r0 do_package: QA Issue: klish: Files/directories were installed but not shipped in any package:
  /usr/lib/clish_plugin_clish.so
```

I'll add full path  to that so file into `FILES_${PN}`

MuteSpirit replied:
fixed

kisyan89 commented on recipes-mmx/mmx-cli/mmx-cli.bb:
```diff
@@ -0,0 +1,93 @@
+# Copyright (c) 2013-2021 Inango Systems LTD.
+#
+# Author: Inango Systems LTD. <support@inango-systems.com>
+# Creation Date: 20 Feb 2021
```
Can you please tell, if Copyright string (line 5) should mention only year 2021 - per the creation date of this file ?

MuteSpirit replied:
Good question.
I don't know
I've just used the same 1st list of Copyright for all MMX files published in OpenSource
There is no strict rule for that
@vicryab @nmerinov Could you please say you opinion about this question?

kisyan89 commented on recipes-mmx/mmx-ep/mmx-ep.bb:
```diff
@@ -0,0 +1,120 @@
+    install -d ${D}/usr/lib/lua/mmx
+    install -m 0755 ${S_FEED}/lua/mmx_db_reformat_utils.lua ${D}/usr/lib/lua/mmx/mmx_db_reformat_utils.lua
+    install -m 0755 ${S_FEED}/lua/mmx_db_reformat_perform.lua ${D}/usr/lib/lua/mmx/mmx_db_reformat_perform.lua
+    install -m 0755 ${S_FEED}/lua/mmx_db_get_version.lua ${D}/usr/lib/lua/mmx/mmx_db_reformat_get_version.lua
```
typo or intentional change from `mmx_db_get_version.lua` to `mmx_db_reformat_get_version.lua`

MuteSpirit replied:
This line has been ported  from [original mmx-ep feed](https://github.com/InangoSystems/feed-mmx/blob/f4c6fb722639ea625d3e2a5d094dc925bb3c4bf8/net/mmx-ep/Makefile#L243) without paths modifications
I've rechecked that SVN version of `inango-openwrt/feeds/net/mmx-ep/Makefile` contains the same paths.

kisyan89 replied:
ACK

kisyan89 commented on recipes-mmx/mmx-ep/mmx-ep.bb:
```diff
@@ -0,0 +1,120 @@
+    install -m 0755 ${S_FEED}/lua/mmx_db_reformat_utils.lua ${D}/usr/lib/lua/mmx/mmx_db_reformat_utils.lua
+    install -m 0755 ${S_FEED}/lua/mmx_db_reformat_perform.lua ${D}/usr/lib/lua/mmx/mmx_db_reformat_perform.lua
+    install -m 0755 ${S_FEED}/lua/mmx_db_get_version.lua ${D}/usr/lib/lua/mmx/mmx_db_reformat_get_version.lua
+    install -m 0755 ${S_FEED}/lua/mmx_db_clear.lua ${D}/usr/lib/lua/mmx/mmx_db_reformat_clear.lua
```
typo or intentional change from `mmx_db_clear.lua` to `mmx_db_reformat_clear.lua` ?

MuteSpirit replied:
This line has been ported  from [original mmx-ep feed](https://github.com/InangoSystems/feed-mmx/blob/f4c6fb722639ea625d3e2a5d094dc925bb3c4bf8/net/mmx-ep/Makefile#L244) without paths modifications
I've rechecked that SVN version of `inango-openwrt/feeds/net/mmx-ep/Makefile` contains the same paths.

kisyan89 commented on recipes-tweaks/busybox/busybox_%.bbappend:
```diff
@@ -0,0 +1,71 @@
+# - professional sub-contract and customization services
+#
+################################################################################
+SRC_URI_remove_dunfell += "file://220-add_lock_util.patch"
```
General note - short explanation for these removals is something worth having

MuteSpirit replied:
I've switched `meta-openwrt` layer to `dunfell` branch and this bbappend became unneeded and has been removed

Ranihilator commented on recipes-core/rpcd/files/rpcd.service:
```diff
@@ -0,0 +1,13 @@
+Requires=local-fs.target ubusd.service
+
+[Service]
+ExecStart=/bin/sh -c '@SBINDIR@/rpcd -s /var/run/ubus/ubus.sock -t 30 &'
```
if we using Type=forking, why we don't set PIDFile?
it allows systemd to control the process themselves. And no need to use detach "&" in ExecStart

MuteSpirit replied:
rpcd does not create PID file so we cannot follow that way
```
root@RaspberryPi-Gateway:~# ls -la /proc/243/fd
total 0
dr-x------    2 root     root             0 May  6 14:58 .
dr-xr-xr-x    9 root     root             0 Feb 18 06:53 ..
lr-x------    1 root     root            64 May  6 14:58 0 -> /dev/null
lrwx------    1 root     root            64 May  6 14:58 1 -> socket:[16419]
lrwx------    1 root     root            64 May  6 14:58 2 -> socket:[16419]
lrwx------    1 root     root            64 May  6 14:58 3 -> anon_inode:[eventpoll]
lr-x------    1 root     root            64 May  6 14:58 4 -> pipe:[16437]
l-wx------    1 root     root            64 May  6 14:58 5 -> pipe:[16437]
lrwx------    1 root     root            64 May  6 14:58 6 -> socket:[16438]
```

Ranihilator commented on recipes-core/rpcd/files/rpcd.service:
```diff
@@ -0,0 +1,13 @@
+Requires=local-fs.target ubusd.service
+
+[Service]
+ExecStart=/bin/sh -c '@SBINDIR@/rpcd -s /var/run/ubus/ubus.sock -t 30 &'
```
what is @SBINDIR@???

I think you should use 
Environment="VAR=Value"
ExecStart=/usr/bin/application -some_key $VAR
and set environment some from external side

MuteSpirit replied:
It has been done according similar cases in another recipes
```
../meta-rdk-bsp-emulator/recipes-connectivity/hostapd/hostapd_%.bbappend:86:    sed -i -e 's,@SBINDIR@,${sbindir},g' -e 's,@SYSCONFDIR@,${sysconfdir},g' ${D}${systemd_unitdir}/system/hostapd.service
../meta-rdk-ext/recipes-common/lighttpd-1.4.39/lighttpd_1.4.39.bb:58:   sed -i -e 's,@SBINDIR@,${sbindir},g' \
../meta-rdk-ext/recipes-common/lighttpd-1.4.39/lighttpd/lighttpd.service:6:ExecStartPre=@SBINDIR@/lighttpd -t -f @SYSCONFDIR@/lighttpd.conf
../meta-rdk-ext/recipes-common/lighttpd-1.4.39/lighttpd/lighttpd.service:7:ExecStart=@SBINDIR@/lighttpd -D -f @SYSCONFDIR@/lighttpd.conf
../meta-rdk-ext/recipes-common/lighttpd/lighttpd_1.4.53.bb:67:  sed -i -e 's,@SBINDIR@,${sbindir},g' \
../meta-virtualization/recipes-extended/irqbalance/irqbalance/irqbalanced.service:6:ExecStart=@SBINDIR@/irqbalance --foreground
../meta-virtualization/recipes-extended/irqbalance/irqbalance.inc:46:        sed -i -e 's,@SBINDIR@,${sbindir},g' ${D}${systemd_unitdir}/system/irqbalanced.service
```

Your idea is good by my opinion but will be fixed it in separate task

Ranihilator commented on recipes-core/ubus/ubus_git.bbappend:
```diff
@@ -0,0 +1,85 @@
+    # Install systemd unit files
+    install -d ${D}${systemd_unitdir}/system
+    install -m 0644 ${WORKDIR}/ubusd.service ${D}${systemd_unitdir}/system
+        sed -i -e 's,@SBINDIR@,${sbindir},g' \
```
I suggest using export instead sed and use
Environment="VAR=value" in systemd services

also, you can use environment file like this, examplein RDK-B
EnvironmentFile=/etc/device.properties
systemd will use these sequences from the file

MuteSpirit replied:
It has been done according similar cases in another recipes
```
../meta-rdk-bsp-emulator/recipes-connectivity/hostapd/hostapd_%.bbappend:86:    sed -i -e 's,@SBINDIR@,${sbindir},g' -e 's,@SYSCONFDIR@,${sysconfdir},g' ${D}${systemd_unitdir}/system/hostapd.service
../meta-rdk-ext/recipes-common/lighttpd-1.4.39/lighttpd_1.4.39.bb:58:   sed -i -e 's,@SBINDIR@,${sbindir},g' \
../meta-rdk-ext/recipes-common/lighttpd-1.4.39/lighttpd/lighttpd.service:6:ExecStartPre=@SBINDIR@/lighttpd -t -f @SYSCONFDIR@/lighttpd.conf
../meta-rdk-ext/recipes-common/lighttpd-1.4.39/lighttpd/lighttpd.service:7:ExecStart=@SBINDIR@/lighttpd -D -f @SYSCONFDIR@/lighttpd.conf
../meta-rdk-ext/recipes-common/lighttpd/lighttpd_1.4.53.bb:67:  sed -i -e 's,@SBINDIR@,${sbindir},g' \
../meta-virtualization/recipes-extended/irqbalance/irqbalance/irqbalanced.service:6:ExecStart=@SBINDIR@/irqbalance --foreground
../meta-virtualization/recipes-extended/irqbalance/irqbalance.inc:46:        sed -i -e 's,@SBINDIR@,${sbindir},g' ${D}${systemd_unitdir}/system/irqbalanced.service
```

Your idea is good by my opinion but will be fixed it in separate task

Ranihilator commented on recipes-mmx/mmx-ep/mmx-ep.bb:
```diff
@@ -0,0 +1,120 @@
+    install -d ${D}${sbindir}
+    install -m 0755 ${S_FEED}/mmxep_env.sh ${D}${sbindir}
+
+    install -d ${D}/etc/init.d
```
etc -> ${sysconfdir}

export base_prefix = ""
export sysconfdir = "${base_prefix}/etc"

MuteSpirit replied:
fixed

Ranihilator commented on recipes-mmx/mmx-ep/mmx-ep.bb:
```diff
@@ -0,0 +1,120 @@
+    install -d ${D}/etc/mmx/saved-db
+    install -d ${D}/etc/mmx/cand-db
+
+    install -d ${D}/usr/lib/lua/mmx
```
/usr/lib -> ${libdir}

export exec_prefix = "/usr"
export libdir = "${exec_prefix}/lib"

MuteSpirit replied:
fixed

Ranihilator commented on recipes-core/procd-base-files/procd-base-files_git.bb:
```diff
@@ -0,0 +1,108 @@
+
+do_install() {
+    install -d ${D}${sysconfdir}
+    install -m 0755 ${BF}/etc/rc.common ${D}${sysconfdir}/rc.common
```
etc -> ${sysconfdir}

export base_prefix = ""
export sysconfdir = "${base_prefix}/etc"

MuteSpirit replied:
fixed

Ranihilator commented on recipes-core/procd-base-files/procd-base-files_git.bb:
```diff
@@ -0,0 +1,108 @@
+    install -m 0755 ${BF}/etc/rc.common ${D}${sysconfdir}/rc.common
+
+    install -d ${D}${base_libdir}/functions
+    install -m 0644 ${BF}/lib/functions/service.sh ${D}${base_libdir}/functions/service.sh
```
/ib -> ${base_libdir}

export base_prefix = ""
export base_libdir = "${base_prefix}/lib"

MuteSpirit replied:
fixed

Ranihilator commented on recipes-mmx/libing-gen-utils/libing-gen-utils.bb:
```diff
@@ -0,0 +1,107 @@
+do_install() {
+    oe_runmake install DESTDIR=${D} LUAPATH=${LUAPATH}
+
+    (cd ${D}/usr/lib && mv ${PN}.so ${PN}.so.${PV})
```
/usr/lb -> ${libdir}

MuteSpirit replied:
fixed

Ranihilator commented on recipes-mmx/libmmx-backapi/libmmx-backapi.bb:
```diff
@@ -0,0 +1,102 @@
+do_install() {
+    oe_runmake install DESTDIR=${D}
+
+    (cd ${D}/usr/lib && mv ${PN}.so ${PN}.so.${PV})
```
/usr/lb -> ${libdir}

MuteSpirit replied:
fixed

Ranihilator commented on recipes-mmx/libmmx-frontapi/libmmx-frontapi.bb:
```diff
@@ -0,0 +1,105 @@
+do_install() {
+    oe_runmake install DESTDIR=${D} LUAPATH=${LUAPATH}
+
+    (cd ${D}/usr/lib && mv ${PN}.so ${PN}.so.${PV})
```
/usr/lb -> ${libdir}

MuteSpirit replied:
fixed

Commit:
README.md: fix typo Grand -> Grant

Signed-off-by: Ivan Efimov <i.efimov@inango-systems.com>

Commit:
klish: fetch sources from tar.gz instead of tar.xz

Yocto Dunfell has some issue which does not allow to use tar.xz as
sources tarball.
During clean build "do_unpack" task prints error that "xz not found"
And that occur even if "xz-native" is in DEPENDS
2nd run of build will be success but it's just workaround.

Replace tar.xz with tar.gz in klish SRC_URI

Commit:
meta-openwrt: switch to "dunfell" branch

meta-openwrt has separate branch for support Yocto 3.1 Dunfell

Switch to that branch to avoid need of adoption "master" branch
to Yocto 3.1

Delete busybox bbappend because it only removes 2 patches non
applicable on meta-openwrt "master" branch but works smoothy
on "dunfell" one

Adopt path to LuCI which fix luci parser library build

LuCI lib base is absent now, install modules/lua-base/luasrc subdir
instead of

"meta-openwrt" now has recipe "liblucihttp".
So "lucihttp" recipe of current meta layer is not needed more.

Replace luci dependency: lucihttp -> liblucihttp

Signed-off-by: Ivan Efimov <i.efimov@inango-systems.com>

Commit:
ubus bbappend: add package "ubusd"

Both meta-amx and meta-openwrt layers has "ubus" recipes.
But only bbappend in meta-amx add package "ubusd" (U-Bus daemon)
into it.

To keep this behavior and allow recipes like prplmesh explicitly
add "ubusd" as runtime dependency the "ubusd" binary and symlink
has been moved into "ubusd" package

Signed-off-by: Ivan Efimov <i.efimov@inango-systems.com>

Commit:
Replace explicit system paths with bitbake variables

/etc     -> ${sysconfdir}
/usr/lib -> ${libdir}
/lib     -> ${base_libdir}

Signed-off-by: Ivan Efimov <i.efimov@inango-systems.com>

Commit:
Switch MMX recipes to last commits after merging their PRs

The set of MMX packages repositories had PRs related to
support MMX on RDKB

After merge next PR the SRC_URI and SRCREV of consequent recipes
has been updated
    https://github.com/InangoSystems/luaxml/pull/2
    https://github.com/InangoSystems/libmmx-frontapi/pull/2
    https://github.com/InangoSystems/libing-gen-utils/pull/2
    https://github.com/InangoSystems/libmmx-backapi/pull/1

Signed-off-by: Ivan Efimov <i.efimov@inango-systems.com>

kisyan89 commented on recipes-extended/luci/luci_git.bbappend:
```diff
@@ -0,0 +1,104 @@
+################################################################################
+#
+# luci_git.bb
```
incorrect name, should be bbappend
I think I saw also not correct name inside liblucihttp_git.bbappend

MuteSpirit replied:
fixed
